File: standardnaast-backend/src/main/java/standardNaast/backend/service/BenevolatService.java
package standardNaast.backend.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import standardNaast.backend.dto.BenevolatCreateUpdateDto;
import standardNaast.backend.dto.BenevolatDto;
import standardNaast.backend.dto.MemberBenevolatSummaryDto;

import java.time.LocalDate;
import java.util.List;

public interface BenevolatService {

    Page<BenevolatDto> searchBenevolats(Long personId, LocalDate startDate, LocalDate endDate, Pageable pageable);

    List<BenevolatDto> getBenevolatsByPerson(Long personId);

    List<BenevolatDto> getBenevolatsByPersonAndSeason(Long personId, String seasonId);

    MemberBenevolatSummaryDto getMemberBenevolatSummary(Long personId, String seasonId);

    BenevolatDto getBenevolatById(Long id);

    BenevolatDto createBenevolat(BenevolatCreateUpdateDto dto);

    BenevolatDto updateBenevolat(Long id, BenevolatCreateUpdateDto dto);

    void deleteBenevolat(Long id);
}

File: standardnaast-ng/src/app/core/services/benevolat.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Benevolat, BenevolatCreateUpdate, MemberBenevolatSummary } from '../models/benevolat.model';
import { Page } from '../models/page.model';

@Injectable({
  providedIn: 'root'
})
export class BenevolatService {
  private readonly apiUrl = `${environment.apiUrl}/benevolats`;

  constructor(private http: HttpClient) {}

  searchBenevolats(personId?: number, startDate?: string, endDate?: string, page = 0, size = 20): Observable<Page<Benevolat>> {
    let params = new HttpParams()
      .set('page', page.toString())
      .set('size', size.toString());

    if (personId) params = params.set('personId', personId.toString());
    if (startDate) params = params.set('startDate', startDate);
    if (endDate) params = params.set('endDate', endDate);

    return this.http.get<Page<Benevolat>>(this.apiUrl, { params });
  }

  getBenevolatsByPerson(personId: number): Observable<Benevolat[]> {
    return this.http.get<Benevolat[]>(`${this.apiUrl}/person/${personId}`);
  }

  getMemberBenevolatSummary(personId: number, seasonId?: string): Observable<MemberBenevolatSummary> {
    let params = new HttpParams();
    if (seasonId) params = params.set('seasonId', seasonId);

    return this.http.get<MemberBenevolatSummary>(`${this.apiUrl}/person/${personId}/summary`, { params });
  }

  getBenevolatById(id: number): Observable<Benevolat> {
    return this.http.get<Benevolat>(`${this.apiUrl}/${id}`);
  }

  createBenevolat(dto: BenevolatCreateUpdate): Observable<Benevolat> {
    return this.http.post<Benevolat>(this.apiUrl, dto);
  }

  updateBenevolat(id: number, dto: BenevolatCreateUpdate): Observable<Benevolat> {
    return this.http.put<Benevolat>(`${this.apiUrl}/${id}`, dto);
  }

  deleteBenevolat(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
